### PR TITLE
Refactor cohort list component to support fetching forms from the backend

### PR DESCRIPTION
**Pull request by samuelmale**

### CODE REVIEW

File: packages/esm-covid-app/src/views/dashboard/patient-list-tabs/covid-patient-list-tabs.component.tsx
Comment (ebambo) [inline packages/esm-covid-app/src/views/dashboard/patient-list-tabs/covid-patient-list-tabs.component.tsx:1]: Translations boss

File: packages/esm-commons-lib/src/components/patient-list-cohort/helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+export function consolidatatePatientMeta(rawPatientMeta, form, config: PatientMetaConfig) {
+  const { isDynamicCohort, location, encounterType, launchableFormProps, moduleName, addPatientToListOptions } = config;
+  const patientUuid = !isDynamicCohort ? rawPatientMeta.patient.uuid : rawPatientMeta.person.uuid;
+  return {
+    timeAddedToList: !isDynamicCohort ? moment(rawPatientMeta.startDate).format('LL') : null,
+    waitingTime: !isDynamicCohort ? moment(rawPatientMeta.startDate).fromNow() : null,
+    location: location && location.name,
+    phoneNumber: '0700xxxxxx',
+    hivResult: 'None',
```
[the remark this answers is not in the document]
Comment (samuelmale) [reply packages/esm-commons-lib/src/components/patient-list-cohort/helpers.tsx:82]: Nice that you've peeked at this, yes it has always been hardcoded, you probably don't recall but you asked the we keep it like that from the initial implementation of the widget.

**Comment by samuelmale** [comment]
@ebamboI appreciate your feedback on this pull request. The primary focus of this PR is to adapt the widget to fetch forms from the backend. The current state of the widget is bloated, convoluted, and difficult to maintain. Due to time constraints, as it was urgently required to work by @MayanjaXL, I could only introduce a helper file and migrate code to simplify the widget. Essentially, I was rearranging the code.

We have plans to refactor this component, possibly utilizing our modern ETL tools, which we briefly discussed yesterday. In summary, your review is valid, and I had similar concerns in mind as I was examining the code.

**Review by pirupius — DISMISSED** [review]
@samuelmale there's a minor name change we can make but if it can wait then let's get this in cause it's essential to other work. Hopefully the rest of the fixes come later on 😞

File: packages/esm-commons-lib/src/components/patient-list-cohort/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+}) => {
+  const [actionText, setActionText] = useState(launchableForm.actionText);
+  const [encounterUuid, setEncounterUuid] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const continueEncounterActionText = launchableForm.actionText || 'Continue encounter ';
+
+  useEffect(() => {
+    if (launchableForm.editLatestEncounter && encounterType && !encounterUuid) {
+      setIsLoading(true);
+      fetchPatientLastEncounter(patientUuid, encounterType).then((lastHtsEncounter) => {
```
[the remark this answers is not in the document]
Comment (samuelmale) [reply packages/esm-commons-lib/src/components/patient-list-cohort/helpers.tsx]: Done

**Review by pirupius — APPROVED** [review]

**Pull request merged**